ios/RNSScreenStack.mm

```diff
@@ -471,11 +505,20 @@ - (void)setPushViewControllers:(NSArray<UIViewController *> *)controllers
       NSMutableArray *newControllers = [NSMutableArray arrayWithArray:controllers];
       [newControllers removeLastObject];
       [_controller setViewControllers:newControllers animated:NO];
+#ifdef RN_FABRIC_ENABLED
+      auto screenController = (RNSScreen *)top;
+      [screenController resetViewToScreen];
+#endif
       [_controller pushViewController:top animated:YES];
     } else {
       // don't really know what this case could be, but may need to handle it
       // somehow
-      [_controller setViewControllers:controllers animated:NO];
+#ifdef RN_FABRIC_ENABLED
+      BOOL shouldAnimate = YES;
+#else
+      BOOL shouldAnimate = NO;
+#endif
+      [_controller setViewControllers:controllers animated:shouldAnimate];
```

Comment:
I think we removed the `shouldAnimate` since it was not needed at all (see https://github.com/software-mansion/react-native-screens/blob/ff617635aeb417a7742337aec6b8aa3f05e2f09b/ios/RNSScreenStack.mm#L449)

Reply:
[e739f8f](https://github.com/software-mansion/react-native-screens/pull/1416/commits/e739f8f9f9d1a2ccec4357b49b8d18f510cb3814)